src/index.js

```diff
@@ -122,7 +126,20 @@ class PhotoshopAPI {
         ApiKeyAuth: { value: apiKey }
       },
       usePromise: true
-    })
+    }
+
+    if (options !== undefined && options.useSwaggerFetch) {
+      // Do nothing because this overrides customized fetch
+      console.log('Using swagger fetch')
+    } else if (options !== undefined && options.userFetch !== undefined) {
+      swaggerOptions.userFetch = options.userFetch
+      console.log('Using custom fetch')
```

Comment:
I would try to add some tests for the logic you added that is not covered yet by unit tests